File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(tmp, idx):
        nonlocal answer
        if idx == len(numbers):
            if tmp == target:
                answer += 1
            return

        dfs(tmp+numbers[idx], idx+1)
        dfs(tmp-numbers[idx], idx+1)
    
    dfs(0, 0)
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    
    public int solution(int[] numbers, int target) {
        dfs(0, 0, target, numbers);
        return answer;
    }
    
    public void dfs(int idx, int tmp, int target, int[] numbers) {
        if(idx == numbers.length) {
            if(tmp == target) {
                answer++;
            }
        } else {
            dfs(idx+1, tmp+numbers[idx], target, numbers);
            dfs(idx+1, tmp-numbers[idx], target, numbers);
        }
    }
}